Esri/arcgis-maps-sdk-swift-samples
# [Update] Add changes for callout in `Add dynamic entity layer`

## Description

This PR updates `Add dynamic entity layer` in `Layers` category.
[URL to README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/Ting/UpdateDynamicEntitySample/Shared/Samples/Add%20dynamic%20entity%20layer/README.md)

## Linked Issue(s)

- `swift/issues/4688`
- `common-samples/pull/3882`

## How To Test

1. Run the sample, all existing features work
2. Tap on a vehicle. A callout should display
3. When the vehicle moves, the callout should follow along
4. When the observations are purged, the callout should be dismissed

## Screenshots

![](https://raw.githubusercontent.com/Esri/arcgis-maps-sdk-swift-samples/9bdeccd2cc982aec60dda0e1a65fed8ba6d62631/Shared/Samples/Add%20dynamic%20entity%20layer/add-dynamic-entity-layer-settings.png)

## To Discuss

- The default values for speed and heading are `nan`. Very occasionally it causes the callout subview to give a warning in the console
> Bound preference SizePreferenceKey tried to update multiple times per frame.

If the default values are set to zero I don't see this warning. Shall we make them zero?

- If I don't reset the placement when observations are purged, the callout would stay in the old place when new observations are generated. Thus I passed a binding to the settings view.
